src/PropelQuery/Schema/PropelSchemaTableFinder.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace ArchitectureSniffer\PropelQuery\Schema;
+
+use ArchitectureSniffer\Module\Transfer\ModuleTransfer;
+use ArchitectureSniffer\PropelQuery\Schema\Transfer\PropelSchemaTableTransfer;
+use Zend\Config\Reader\ReaderInterface;
+
+class PropelSchemaTableFinder implements PropelSchemaTableFinderInterface
+{
+    /**
+     * @var \Zend\Config\Reader\ReaderInterface
+     */
+    protected $reader;
+
+    /**
+     * @param \Zend\Config\Reader\ReaderInterface $reader
+     */
+    public function __construct(ReaderInterface $reader)
+    {
+        $this->reader = $reader;
+    }
+
+    /**
+     * @param \ArchitectureSniffer\Module\Transfer\ModuleTransfer $moduleTransfer
+     *
+     * @return \ArchitectureSniffer\PropelQuery\Schema\Transfer\PropelSchemaTableTransfer[]
+     */
+    public function getTablesByModule(ModuleTransfer $moduleTransfer): array
+    {
+        $tableTransfers = [];
+
+        $moduleName = $moduleTransfer->getModuleName();
+
+        foreach ($moduleTransfer->getSchemaPaths() as $schemaPath) {
+            $schema = $this->reader->fromFile($schemaPath);
+
+            $tableTransfers = $this->getTables(
+                $schema,
+                $moduleName,
+                $tableTransfers
+            );
+        }
+
+        return $tableTransfers;
+    }
+
+    /**
+     * @param array $table
+     *
+     * @return bool
+     */
+    protected function isTableMain(array $table): bool
```

Comment:
This name doesn't make clear that you check if the current table is defined in the current module. When a table has the primary key definition the module where this file is found is the owner of the table.

IMO the method name should be more like `isOwnerOfTable` or sth like that.